### PR TITLE
(SIMP-4438) sssd: Troubleshoot AD issues

### PR DESCRIPTION
**Pull request by jeefberkey**
* Some template and concat changes to make debugging the module easier
  * Add headers to more easily diagnose where to stick params
  * Order the concat sections
* Add some missing params from sssd::provider::ad
  * default_shell
  * dyndns_update
  * fallback_homedir
  * krb5_realm
  * krb5_store_password_if_offline

### CODE REVIEW

**Comment by jeefberkey** [comment]
:white_check_mark: https://gitlab.com/simp/pupmod-simp-sssd/pipelines/17972170

File: manifests/domain.pp [hunk 1]
```diff
@@ -75,7 +75,7 @@
   Optional[String]                           $description                  = undef,
   Integer[0]                                 $min_id                       = simplib::lookup('simp_options::uid::min', { 'default_value' => pick(fact('login_defs.uid_min'), 1000) }),
   Integer[0]                                 $max_id                       = simplib::lookup('simp_options::uid::max', { 'default_value' => pick(fact('login_defs.uid_max'), 0 ) }),
-  Boolean                                    $enumerate                    = false,
+  Optional[Boolean]                          $enumerate                    = false,
```
Comment (lnemsick-simp) [inline manifests/domain.pp]: Optional when not undef doesn't really make sense?

**Comment by jeefberkey** [comment]
:white_check_mark: https://gitlab.com/simp/pupmod-simp-sssd/pipelines/18059171

**Comment by jeefberkey** [comment]
Un-WIPping this PR because the stated bugs are fixed. There will be an acceptance test coming that will set up an AD server in a Beaker environment and use this module to connect to it with sssd, realmd, etc.

**Pull request merged**